esp/esp/program/modules/handlers/teacherbigboardmodule.py

```diff
@@ -0,0 +1,163 @@
+import datetime
+import operator
+import subprocess
+
+from django.db.models.aggregates import Count, Max, Min
+from django.db.models.query import Q
+
+from argcache import cache_function_for
+from esp.program.models import ClassSubject
+from esp.program.modules.base import ProgramModuleObj, needs_admin, main_call
+from esp.users.models import Record
+from esp.utils.decorators import cached_module_view
+from esp.utils.web import render_to_response
+from esp.program.modules.handlers.bigboardmodule import BigBoardModule
+
+
+class TeacherBigBoardModule(ProgramModuleObj):
+
+    @classmethod
+    def module_properties(cls):
+        return {
+            "admin_title": "Teacher Registration Big Board",
+            "link_title": "Watch incoming teacher registrations",
+            "module_type": "manage",
+            "seq": 11
+            }
+
+    class Meta:
+        proxy = True
+        app_label = 'modules'
+
+    @main_call
+    @needs_admin
+    def teacherbigboard(self, request, tl, one, two, module, extra, prog):
+        """Loads a few numbers about teacher registration.
+
+        This may be loaded by a bunch of admins at once and refreshed quickly,
+        at a time when the website is already slow, so it needs to be really
+        fast.
+        """
+        # Most of the numbers on the page are rendered from this list, which
+        # should consist of pairs (description, number)
+        hours_stats = self.static_hours(prog)
+        numbers = [
+            ("teachers registering in the last 10 minutes",
+             self.num_active_users(prog)),
+            ("teachers teaching a class",
+             self.num_teachers_teaching(prog)),
+            ("classes registered",
+             self.num_class_reg(prog)),
+            ("classes approved",
+             self.num_class_app(prog)),
+            ("class-hours registered",
+             hours_stats[0]),
+            ("class-student-hours registered",
+             hours_stats[1]),
+            ("teachers checked in today",
+             self.num_checked_in_teachers(prog)),
+        ]
+
+        numbers = [(desc, num) for desc, num in numbers if num]
+
+        timess = [
+            ("number of registered classes", [(1, time) for time in self.reg_classes(prog)]),
+            ("number of approved classes", [(1, time) for time in self.app_classes(prog)]),
+            ("number of teachers teaching", [(1, time) for time in self.teach_times(prog)]),
+        ]
+
+        graph_data, start = BigBoardModule.make_graph_data(timess)
+
+        class_hours, student_hours = self.get_hours(prog)
+
+        hourss = [
+            ("number of registered class-hours", class_hours),
+            ("number of registered class-student-hours", student_hours),
+        ]
+
+        hours_data, _ = BigBoardModule.make_graph_data(hourss)
```

Comment:
Do you mean it *will* be the same? Because that's what I was assuming. I'm not sure how you'd have a teacher registered without any hours also registered, and those things should start at the same time.

Reply:
Ah, yeah, sorry, will be the same.